math/eigen/include/algebra/math/impl/eigen_transform3.hpp

```diff
@@ -153,8 +153,14 @@ struct transform3 {
     _data_inv = _data.inverse();
   }
 
+  /** Default constructor: set contents to identity matrices */
+  ALGEBRA_HOST_DEVICE
+  transform3() {
+    _data.setIdentity();
+    _data_inv.setIdentity();
+  };
+
   /** Default contructors */
-  transform3() = default;
   transform3(const transform3 &rhs) = default;
   ~transform3() = default;
```

Comment:
Curiously enough it was these defaulted functions that caught my attention... After noticing them, I was convinced that they should be interfering with CUDA. But they don't.

Then I thought that aha, let's try adding `ALGEBRA_HOST_DEVICE` for the copy-constructor, add a line in one of the CUDA tests that copies such an object, and then we should get the same warning that Eigen treats us to on line:

https://gitlab.com/libeigen/eigen/-/blob/master/Eigen/src/Core/DenseStorage.h#L225

But that didn't happen either. :confused: So now I'm really not sure when CUDA likes and doesn't like functions to be declared `= default`. :confused:

In either case, the build also works with these defaulted functions not being declared at all. Since in https://github.com/acts-project/algebra-plugins/commit/62b6bb0ae58ece8d06ef12fa2958c99e9e3c38f2#diff-b0cb69d3ce20cba1d0bdd4c6bf0140e40433133e2c36bca83cc77444ab57603e I ended up doing just that, to get rid of a warning.